app/consolidated-view-poc/react/app/LinkedContacts.js

```diff
@@ -0,0 +1,368 @@
+  // Load list of business customers
+  const { data: businessCustomers, loading: loadingBusinessCustomers } = useQuery(
+    GET_BUSINESS_CUSTOMERS,
+    {
+      variables: { sbi },
+      headers: { email },
+      preloaded: preloaded.businessCustomers
+    }
+  )
+
+  const { search: searchBusinessCustomers, results: businessCustomersSearchResults } = useSearch(
+    businessCustomers?.business?.customers,
+    {
+      idField: 'crn',
+      fields: ['firstName', 'lastName', 'crn'],
+      storeFields: ['firstName', 'lastName', 'crn']
+    }
+  )
+
+  // Create `getCustomer` query
+  const [getCustomer, { data: selectedCustomer, loading: loadingSelectedCustomers }] = useLazyQuery(
+    GET_CUSTOMER,
+    {
+      headers: { email },
+      preloaded: preloaded.selectedCustomer
+    }
+  )
+
+  // Execute `getCustomer` to get first customer in list
+  useEffect(() => {
+    if (businessCustomers?.business?.customers[0].crn) {
+      getCustomer({ crn: businessCustomers?.business?.customers[0].crn, sbi })
+    }
+  }, [businessCustomers])
+
+  // Authenticate questions
+  const [
+    getAuthenticationQuestions,
+    { data: authenticationQuestions, loading: loadingAuthenticationQuestions }
+  ] = useLazyQuery(GET_AUTHENTICATE_QUESTIONS, {
+    headers: { email }
+  })
+
+  const [showAuthenticationQuestions, setShowAuthenticationQuestions] = useState(false)
+
+  useEffect(() => {
+    if (
+      (showAuthenticationQuestions && !authenticationQuestions) ||
+      (showAuthenticationQuestions &&
+        authenticationQuestions?.customer?.crn !== selectedCustomer?.customer?.crn)
+    ) {
+      getAuthenticationQuestions({ crn: selectedCustomer?.customer?.crn })
+    }
+  }, [showAuthenticationQuestions])
+
+  const [selectedPermissionIndex, setSelectedPermissionIndex] = useState(0)
+  const loadingLeftColumn = loadingBusinessCustomers
+  const loadingRightColumn = loadingLeftColumn || !selectedCustomer || loadingSelectedCustomers
+
+  // Reset right column scroll position when loading customer
+  const rightColumnRef = useRef(null)
+  useEffect(() => {
+    rightColumnRef.current.scrollTop = 0
+    setShowAuthenticationQuestions(false)
+  }, [loadingSelectedCustomers])
+
+  return html`
+    <div className="container">
+      <div className="column">
+        <div className="search-input">
+          <label>Search</label>
+          <input
+            name="search"
+            placeholder="Enter search term"
+            ...${{ autoComplete: 'off' }}
+            onChange=${(e) => {
+              searchBusinessCustomers(e.target.value)
+            }}
+          />
+        </div>
+        <div className="primary-table">
+          <table>
+            <thead>
+              <tr>
+                <th>CRN</th>
+                <th>First Name</th>
+                <th>Last Name</th>
+              </tr>
+            </thead>
+            ${businessCustomersSearchResults.length
+              ? html`<tbody className="clickable">
```

Comment:
This sort of block would be much easier if we could lean into the full power of React components 🤞 hopefully we'll get there...